portfolio/src/main/webapp/script.js

```diff
@@ -12,20 +12,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function getComments() {
-  fetch('/data').then(response => response.json()).then((data) => {
-    const commentHistoryEl = document.getElementById('comment-container');
-    if (!Array.isArray(data.comments) || !data.comments.length) {
-        data.forEach((line) => {
-        commentHistoryEl.appendChild(createListElement(line.comment));
-        });
+// retrieve comments and show comment text on page
+async function getComments() {
+    var servletURL = "/data";
+    var maxComments = document.getElementById("max-comments").value;
+    if (maxComments != null && maxComments >= 1) {
+        servletURL = `/data?max-comments=${maxComments}`;
     }
-  });
+    const response = await fetch(servletURL);
+    const comments = await response.json();
+	const commentsList = document.getElementById('comment-container');
+	commentsList.innerHTML = '';
+	comments.forEach(comment => {
+		const content = `${comment.comment}`;
+		commentsList.appendChild(createListElement(content));
+    });
+}
+
+// delete all comments
+var response;
+async function deleteComments() {
+    var servletURL = "/delete-data";
+    const response = await fetch(servletURL, {
+        method: "POST"
+    });
+    await getComments();
 }
 
 /** Creates an <li> element containing text. */
 function createListElement(text) {
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerHTML = "- " + text;
   return liElement;
 }
+
+// show terrain map of CMU with Gates building marker and info window on screen using google maps API
+var map;
```

Comment:
Is this declaration needed? It looks like all references are within functions

Reply:
removed!